### ISSUE TITLE
updateData cause full recalculation of technical indicators

### ISSUE THREAD

**Issue by ovresko** [liihuu/KLineChart#203]
- updateData with last price causes chart to do  full recalculation of technical indicators from start ts.

**Comment by liihuu** [comment]
The calculation of most technical indicators requires some intermediate values that have been calculated before, and this part is difficult.

**Comment by kimboox44** [comment]
exactly, since most TA requires known timeperiod, why not create (update shape) when updateData  is used get X last TA values
and instead of 
 calcTechnicalIndicator: (kLineDataList: any[], { params, plots }) where kLineDataList is full timeseries
 calcTechnicalIndicator: (kLinePeriodDataList: any[], { params, plots }) where kLinePeriodDataList is X last timeseries
